images/win/scripts/Installers/Install-JavaTools.ps1

```diff
@@ -87,17 +96,38 @@ function Install-JavaFromAdoptOpenJDK {
     New-Item -ItemType File -Path $javaVersionPath -Name "$Architecture.complete" | Out-Null
 }
 
-$jdkVersions = (Get-ToolsetContent).java.versions
+$jdkVendors = (Get-ToolsetContent).java.vendors.name
+$defaultVendor = (Get-ToolsetContent).java.default_vendor
 $defaultVersion = (Get-ToolsetContent).java.default
 
-foreach ($jdkVersion in $jdkVersions) {
-    Install-JavaFromAdoptOpenJDK -JDKVersion $jdkVersion
+foreach ($jdkVendor in $jdkVendors) {
+
+    $jdkVersionsToInstall = ((Get-ToolsetContent).java.vendors | Where-Object { $_.name -eq $jdkVendor }).versions
+    $isDefaultVendor = $jdkVendor -eq $defaultVendor
+
+    foreach ($jdkVersionToInstall in $jdkVersionsToInstall) {
+        $isDefaultVersion = $jdkVersionToInstall -eq $defaultVersion
+
+        Install-JavaJDK -VendorName $jdkVendor -JDKVersion $jdkVersionToInstall
+
+        if ($isDefaultVendor -and $isDefaultVersion) 
+        {
+            Set-JavaPath -Version $jdkVersionToInstall -VendorName $jdkVendor -Default
+
+        } elseif ($isDefaultVendor -and (-not($isDefaultVersion))) {
+            # Set default vendor ant non-default vendor's version
+            # тot to coflict with Adopt
+            Set-JavaPath -Version $jdkVersionToInstall -VendorName $jdkVendor
+        }
```

Comment:
I suggest to simplify this condition a bit:
```suggestion
        if ($isDefaultVendor) 
        {
            Set-JavaPath -Version $jdkVersionToInstall -VendorName $jdkVendor -Default:($isDefaultVersion)

        }
```

Reply:
Stopped on the nested `if else` implementation as discussed privately.